File: leetcode/java/14.longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs == null || strs.length == 0) return "";
        String prefix = strs[0];
        StringBuilder pre = new StringBuilder(prefix);
        for(int i = 1; i < strs.length; i++){
            if(pre.length() == 0) return "";
            while(pre.length() > 0){
                int pos = strs[i].indexOf(pre.toString()); 
                if(pos != 0) pre.setLength(pre.length() - 1);
                else break; 
            }
        }
        return pre.toString();
    }
}

File: leetcode/golang/14.longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
    if len(strs) == 0 {
        return ""
    }
    var res []byte
    i := 0
    for i < len(strs[0]) {
        c := strs[0][i]
        for _, v := range strs {
            if len(v) == 0 {
                return ""
            }
            if len(v) <= i || c != v[i] {
                return string(res)
            }
        }
        i++
        res = append(res, c)
    } 
    return string(res)
}